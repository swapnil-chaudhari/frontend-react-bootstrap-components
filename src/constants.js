export const FILTER_PERIODS = {
    YEAR: 365,
    MONTH: 27,
    WEEK: 6,
    DAYS_90: 90,
};

export const SEARCH_TYPES = {
    ARTISTS: 'artists',
    TRACKS: 'tracks'
};

export const KEYS = {
    ENTER: 13,
};
