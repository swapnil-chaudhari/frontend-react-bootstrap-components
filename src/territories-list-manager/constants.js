export const TERRITORIES_KEYS = {
    ID: 'orchId',
    NAME: 'territoryName',
    CONTINENT: 'continent'
};
