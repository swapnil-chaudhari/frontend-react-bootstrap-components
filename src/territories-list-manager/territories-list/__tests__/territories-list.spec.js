import React from 'react';
import renderShallow from 'render-shallow';
import { findWithClass } from 'react-shallow-testutils';
import TerritoriesList from '../territories-list';
import { TERRITORIES_KEYS } from '../../constants';
import noop from 'src/utils/noop';

describe('<TerritoriesList>', () => {
    const classes = (element) => element.props.className.split(' ');
    const theOtherListLabel = 'NOT Cleared for Sale';
    const formatMessage = msg => msg;
    const messages = {
        emptyListLabel: 'empty list label',
        emptyListLabelFrom: 'empty list label from',
        emptyListDescription: 'empty list description'
    };

    describe('when it renders', () => {
        let component;
        const props = {
            theOtherListLabel,
            onCountrySelect: noop,
            territories: [
                { orchId: 14, continent: 'Asia', territoryName: 'Afghanistan' },
                { orchId: 15, continent: 'Africa', territoryName: 'Zimbabwe' },
                { orchId: 16, continent: 'Asia', territoryName: 'Nepal' }
            ],
            selectedCountries: [
                { orchId: 14, continent: 'Asia', territoryName: 'Afghanistan' },
                { orchId: 15, continent: 'Africa', territoryName: 'Zimbabwe' }
            ],
            expandAll: false,
            territoryIdKey: TERRITORIES_KEYS.ID,
            territoryNameKey: TERRITORIES_KEYS.NAME,
            continentNameKey: TERRITORIES_KEYS.CONTINENT,
            formatMessage,
            messages
        };

        beforeAll(() => {
            component = renderShallow(<TerritoriesList { ...props } />).output;
        });

        test('renders the correct className', () => {
            expect(classes(component)).toContain('TerritoriesList');
        });

        test('should render continents and countries', () => {
            expect(component).toMatchSnapshot();
        });

        test('should contain 2 continents', () => {
            const children = findWithClass(
                component,
                'TerritoriesList'
            ).props.children;

            expect(children).toHaveLength(2);
        });
    });

    describe('when it renders without countries', () => {
        let component;
        const props = {
            theOtherListLabel,
            onCountrySelect: noop,
            territories: [],
            selectedCountries: [],
            territoryIdKey: TERRITORIES_KEYS.ID,
            territoryNameKey: TERRITORIES_KEYS.NAME,
            continentNameKey: TERRITORIES_KEYS.CONTINENT,
            formatMessage,
            messages
        };

        beforeAll(() => {
            component = renderShallow(<TerritoriesList { ...props } />).output;
        });

        test('renders the correct className', () => {
            expect(classes(component)).toContain('TerritoriesList-empty');
        });

        test('should render empty list when no countries passed', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
