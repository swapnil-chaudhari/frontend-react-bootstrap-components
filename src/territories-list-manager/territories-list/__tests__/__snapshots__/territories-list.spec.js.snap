// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TerritoriesList> when it renders should render continents and countries 1`] = `
<div
  className="TerritoriesList"
>
  <ContinentItem
    countries={
      Array [
        Object {
          "continent": "Asia",
          "isSelected": true,
          "orchId": 14,
          "territoryName": "Afghanistan",
        },
        Object {
          "continent": "Asia",
          "isSelected": false,
          "orchId": 16,
          "territoryName": "Nepal",
        },
      ]
    }
    expandAll={false}
    name="Asia"
    onCountrySelect={[Function]}
    territoryIdKey="orchId"
    territoryNameKey="territoryName"
  />
  <ContinentItem
    countries={
      Array [
        Object {
          "continent": "Africa",
          "isSelected": true,
          "orchId": 15,
          "territoryName": "Zimbabwe",
        },
      ]
    }
    expandAll={false}
    name="Africa"
    onCountrySelect={[Function]}
    territoryIdKey="orchId"
    territoryNameKey="territoryName"
  />
</div>
`;

exports[`<TerritoriesList> when it renders without countries should render empty list when no countries passed 1`] = `
<div
  className="TerritoriesList-empty"
>
  <p>
    empty list label
  </p>
  <p>
    empty list label from
    <b>
       NOT Cleared for Sale 
    </b>
    empty list description
  </p>
</div>
`;
